### ISSUE TITLE
Timecode Feature Request

### ISSUE THREAD

**Issue by ericpardee** [lhc70000/iina#1003]
**System and IINA version:**

- macOS 10.12.6
- IINA 0.0.12 Build 37

**Expected behavior:**
Feature Request.
As I've replaced QuickTime with IINA, I miss timecode in QuickTime .mov files for work. QuickTime's can have a Timecode Track:
[https://www.dropbox.com/s/p7n6ozvzl0jgabz/Screenshot%202017-08-31%2014.19.11.png?dl=0](url)

When "Show timecode when available" is checked in Preferences:
[https://www.dropbox.com/s/l2cytiljdu1d5fl/Screenshot%202017-08-31%2014.29.09.png?dl=0](url)

You can see the timecode track:
[https://www.dropbox.com/s/x1ub95eqyifrxdk/Screenshot%202017-08-31%2014.19.20.png?dl=0](url)

Great project.

**Comment by bizmar** [comment]
Quickteme 7 is going away with the next OS version. There is no alternative player (to my knowledge, and is free) that can display embedded timecode or just the current play position in the timecode format, or in a "current frame" format. This is very important for almost every video professional using a Mac, we all still have QT7 because of it.

**Comment by mwjburton** [comment]
Was about to post the exact same feature request. With Quicktime 7.0 soon to be gone, the one feature that IINA currently lacks is the timecode display. To have this display in IINA as it has in QT7 would be beyond amazing.

**Comment by melmatsuoka** [comment]
+1 on this feature. I pay a crapload of money for a Telestream Switch license just so I can scrub through .mov files and see the timecode numbers (I work in Visual effects and video post production, so timecode is a non-negotiable requirement in a media player). I'm shocked that the scrubbing performance of IINA is so much better than Switch, and this app is free. Adding timecode support would an absolute godsend.

**Comment by hutsonh** [comment]
+1 please — w've been hunting for a good playback app that can display timecode tracks.

**Comment by brimestoned** [comment]
I just want to share, I've added timecode feature to my fork on iina. Looking for Beta testers. Eventually would like to submit a pull request.

[https://app.box.com/shared/static/bkwcgu4jsmp4c8g8y2ux1i7kakjlhbnw.png](url)

**Comment by mwjburton** [comment]
Amazing. Can beta test when you're ready, this is huge for me.

**Comment by ericpardee** [comment]
> I just want to share, I've added timecode feature to my fork on iina. Looking for Beta testers. Eventually would like to submit a pull request.
> 
> ![alt text](https://camo.githubusercontent.com/040b38e2cf82a15c441838d6f83622b93f4d955c/68747470733a2f2f6170702e626f782e636f6d2f7368617265642f7374617469632f626b77636775346a736d703463386738793275783169376b616b6a6c68626e772e706e67)

What up Ted! Push your code to your fork and I'll test it!

**Comment by brimestoned** [comment]
Download the Beta here. 

```
https://hangar.pilotware.com/share.cgi?ssid=0hPnmDf
Passwd: beta
Contact: ted @ atlasdigital . tv
```

**Comment by hutsonh** [comment]
Excellent, thank you!

(We use Pilotware at work and contract with Atlas frequently!)



—HH

**Comment by theonlygusti** [comment]
Hi I am on IINA 1.0.4

How do I show the exact timestamp?

I am trying to create subtitle files.

**Comment by mwjburton** [comment]
> Download our Beta/Fork here.
> 
> ```
> https://hangar.pilotware.com/share.cgi?ssid=07QGJsv
> Password: beta
> let email = "ted" + "@" + "atlasdigital.tv"
> ```

Have tested the latest available app from this link - built 4th Dec. It looks great. Frame Stepping is not always updating each frame count display though. Approx. 3x every second the count does not update when frame stepping forward or back. Picture updates, TC does not, but comes back into sync on next frame step. Have been testing with 24fps DNx115 Quicktime .MOV files.

Any chance this feature will get merged into IINA?

**Comment by AppleFire** [comment]
Is IILA able to display the SMPTE timecode track yet from QT files? This feature request has been made by multiple parties since 2017. I would like to add to this request. I am using version 1.1.1 and I do not see the ability to display SMPTE timecode.

**Comment by rsomani95** [comment]
+1 for Timecode support

**Comment by theonlygusti** [comment]
It has been such a long time since this issue was created that GitHub now supports images directly.

Here is the image of Quicktime 7's "timecode" that the OP posted:

![Ksalug4I](https://user-images.githubusercontent.com/10684786/104125180-0ec04180-534d-11eb-861c-3acff9d7d4e1.png)

IINA must add support for this already.

**Comment by marcandregray** [comment]
+1 would love a frame counter in IINA

**Comment by deeeeekun** [comment]
It's been a while but I am on ver. 1.3.0 on macOS and the timestamp can be displayed in milliseconds. Just right-click on the timestamp on either side of the progress bar and it will show you the precisions option.

**Comment by mwjburton** [comment]
While that can be useful, it's not a timecode display with frame count which is what we really need. Adding the option to that right click menu would make sense and have it be a sticky setting.

**Comment by The-Johnson** [comment]
Adding one more plead for this feature, a hh:mm:ss:ff timecode is a must for anything professional. 

@nodingneu My dude, please check out a subtitling software like AegiSub or SubtitleEdit for your needs, don't be writing timecodes manually, that is very, extremely, unnecessary.

**Comment by jackthegiantkiller** [comment]
Yes, we _really_ need timecode and frame support for this app, or have I missed this function being added? Such a great app, this would really make it useful for video and animation professionals.
